lms/djangoapps/completion/api/v1/tasks.py

```diff
@@ -0,0 +1,27 @@
+from celery import current_task, task  # pylint: disable=no-name-in-module
+from celery.exceptions import RetryTaskError  # pylint: disable=no-name-in-module, import-error
+from celery.states import FAILURE, RETRY, SUCCESS  # pylint: disable=no-name-in-module, import-error
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from student.models import CourseEnrollmentManager
+from lms.djangoapps.completion.utils import GenerateCompletionReport
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+
+@task(default_retry_delay=5, max_retries=5)
+def generate_report(course_id):
+
+    try:
+        course_key = CourseKey.from_string(course_id)
+    except InvalidKeyError:
+        return False
+
+    users = CourseEnrollmentManager().users_enrolled_in(course_key)
+    completion_report = GenerateCompletionReport(users, course_key)
```

Comment:
I don't remember if we talked about this before, but I saw you have some calls to configuration_helpers on GenerateCompletionReport class methods. Remember we don't have microsite or django site on a celery function, so this could lead us to unexpected results if we are relying on reading some microsite settings when building the report